src/pyk/kast/outer.py

```diff
@@ -1154,14 +1154,30 @@ def greatest_common_subsort(self, sort1: KSort, sort2: KSort) -> KSort | None:
         return None
 
     # Sorts like Int cannot be injected directly into sort K so they are embedded in a KSequence.
-    def add_ksequence_under_kequal(self, kast: KInner) -> KInner:
-        def _add_ksequence_under_kequal(kast: KInner) -> KInner:
-            if type(kast) is KApply and kast.label.name == '_==K_':
-                return KApply('_==K_', [KSequence(arg) for arg in kast.args])
-            else:
+    def add_ksequence_under_k_productions(self, kast: KInner) -> KInner:
+        def _add_ksequence_under_k_productions(kast: KInner) -> KInner:
+            def _is_k(kast: KInner) -> bool:
+                if type(kast) is not KVariable:
+                    return False
+                if kast.sort is None:
+                    return True
+                if kast.sort.name == 'K':
+                    return True
+                return False
```

Comment:
Can we just use `self.sort(kast)` instead? So we could say:

```suggestion
            def _is_k(kast: KInner) -> bool:
                return self.sort(kast) == KSort('K')
```

In which case, we don't need the helper probably?

Reply:
Changed to use `self.sort(kast.args[i]) == KSort('K')`